pkg/objstore/s3/s3.go

```diff
@@ -149,6 +149,7 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 			DualStack: true,
 		}).DialContext,
 		MaxIdleConns:          100,
+		MaxIdleConnsPerHost:   100,
```

Comment:
Since the default in our code-base is already this is probably OK. We should probably opt to have this value hard-coded, not sure if anyone will want to change it. Also, I wonder how this will affect performance. There are most likely way too many variables to have objective data.